packages/client/src/hooks/useAccount.js

```diff
@@ -0,0 +1,85 @@
+import { mutate, query, tx } from "@onflow/fcl";
+import {
+  GET_USER_FLOW_BALANCE,
+  GET_USER_FUSD_BALANCE,
+  GET_USER_USDC_BALANCE,
+  SEND_TOKENS_TO_TREASURY,
+} from "flow";
+import { COIN_TYPE_TO_META } from "constants/maps";
+import { COIN_TYPES } from "constants/enums";
+import { REGULAR_LIMIT } from "constants/constants";
+
+const COIN_TYPE_LIST = [COIN_TYPES.FLOW, COIN_TYPES.FUSD, COIN_TYPES.USDC];
+
+export default function useAccount() {
+  const doSendTokensToTreasury = async (treasuryAddr, amount, coinType) => {
```

Comment:
I'm wondering if we should allow only signers to add tokens to the treasury?
@jacksonConrad @amyliu6 @linxiao-li 
If we decide to change the logic around this, we will tackle it in another task, this is just for discussion purposes only 😄

Reply:
we've discussed this with Jose, and the intended functionality is that only signers can add a net-new `Vault` to the Treasury, but ANYONE can send tokens to an existing Vault in the treasury.